### PR TITLE
Enabling Basic Auth configured at runtime

### PR DESCRIPTION
**Pull request by andreaTP**
Opening for visibility, I'll finish up this one at some point next week.

### CODE REVIEW

**Review by EricWittmann — APPROVED** [review]
LGTM but would like @carlesarnal to also review

File: auth/src/main/java/io/apicurio/common/apps/auth/authn/AppAuthenticationMechanism.java [hunk 1]
```diff
@@ -71,16 +64,22 @@
...
     @Info(category = "auth", description = "Enable auth", availableSince = "0.1.18-SNAPSHOT", registryAvailableSince = "2.0.0.Final", studioAvailableSince = "1.0.0")
-    boolean authEnabled;
+    boolean oidcAuthEnabled;
 
+    // back to fake auth and use another property
     @Dynamic(label = "HTTP basic authentication", description = "When selected, users are permitted to authenticate using HTTP basic authentication (in addition to OAuth).", requires = "apicurio.authn.enabled=true")
     @ConfigProperty(name = "apicurio.authn.basic-client-credentials.enabled", defaultValue = "false")
     @Info(category = "auth", description = "Enable basic auth client credentials", availableSince = "0.1.18-SNAPSHOT", registryAvailableSince = "2.1.0.Final", studioAvailableSince = "1.0.0")
-    Supplier<Boolean> fakeBasicAuthEnabled;
+    Supplier<Boolean> basicClientCredentialsAuthEnabled;
```
Comment (carlesarnal) [inline auth/src/main/java/io/apicurio/common/apps/auth/authn/AppAuthenticationMechanism.java:78]: good, much better

File: auth/src/main/java/io/apicurio/common/apps/auth/authn/AppAuthenticationMechanism.java [hunk 2]
```diff
@@ -222,17 +236,31 @@ public String getForwardedFor() {
...
+        return credentialTypes;
     }
 
     @Override
-    public HttpCredentialTransport getCredentialTransport() {
-        return new HttpCredentialTransport(HttpCredentialTransport.Type.AUTHORIZATION, "bearer");
+    public Uni<HttpCredentialTransport> getCredentialTransport(RoutingContext context) {
+        var enabledAuth = selectEnabledAuth();
+        if (enabledAuth != null) {
+            // TODO: was new HttpCredentialTransport(HttpCredentialTransport.Type.AUTHORIZATION, "bearer");
```
Comment (carlesarnal) [inline auth/src/main/java/io/apicurio/common/apps/auth/authn/AppAuthenticationMechanism.java:259]: Should this be changed to honour the configured auth type?

**Review by carlesarnal — APPROVED** [review]
Just a small question, otherwise, lgtm

File: auth/src/main/java/io/apicurio/common/apps/auth/authn/AppAuthenticationMechanism.java (hunk 2 above)
Comment (andreaTP) [reply auth/src/main/java/io/apicurio/common/apps/auth/authn/AppAuthenticationMechanism.java:259]: it is, I just need to remove the TODO comment which is a leftover

Comment (andreaTP) [reply auth/src/main/java/io/apicurio/common/apps/auth/authn/AppAuthenticationMechanism.java]: done

**Review by carlesarnal — APPROVED** [review]

File: auth/src/main/java/io/apicurio/common/apps/auth/authn/AppAuthenticationMechanism.java (hunk 2 above)
Comment (carlesarnal) [reply auth/src/main/java/io/apicurio/common/apps/auth/authn/AppAuthenticationMechanism.java]: oh right, it's using the credential transport from the enabled auth

**Pull request merged**